### DISCUSSION TITLE
Troubleshooting UPS SurePost

### DISCUSSION THREAD

**Discussion by owi2000** [alexeybusygin/ShippingRates#59] [Q&A] [answered]
I have this added to a project via nuget and it's been working pretty well.  We recently signed up to get SurePost rates from UPS, however the rates do not show up in the RateManager.GetRates() returns.  Is there something that needs to be added to the request in order to get the rates?  If not, is there a way to get access to the XML being created that is sent to UPS so I can open a ticket with them?

**Comment by owi2000** [comment]
It turns out the problem is the RequestOption value.  ShippingRates defaults it to "Shop" unless you specify in the UPSProvider constructor the serviceDescription.  Submitting "Rate" into the serviceDescription doesn't work, it results in an error when getting the UPS Ship Code because it's not a valid entry.  So I changed it to "UPS Sure Post" which is in the ToUpsShipCode() function.  Unfortunately this resulted in no returned rates.  This is probably because "UPS Sure Post" is not a valid entry for the RequestOption (which needs to be "Rate").  Unfortunately both values are driven by the same variable - _serviceDescription.  
Hopefully someone can help me with this additional information.

**Reply by owi2000** [reply]
Decided to submit this as a bug, as I don't see a way to make it work as-is.

**Comment by alexeybusygin** [comment, marked as answer]
Solved in v2.1.0.
